src/wallet/wallet.h

```diff
@@ -513,7 +541,23 @@ class CWalletTx
         std::vector<uint256> dummy_vector1; //!< Used to be vMerkleBranch
         std::vector<CMerkleTx> dummy_vector2; //!< Used to be vtxPrev
         char dummy_char; //! Used to be fSpent
-        s >> tx >> hashBlock >> dummy_vector1 >> nIndex >> dummy_vector2 >> mapValue >> vOrderForm >> fTimeReceivedIsTxTime >> nTimeReceived >> fFromMe >> dummy_char;
+        int serializedIndex;
+        s >> tx >> tx_state.hashBlock >> dummy_vector1 >> serializedIndex >> dummy_vector2 >> mapValue >> vOrderForm >> fTimeReceivedIsTxTime >> nTimeReceived >> fFromMe >> dummy_char;
+
+        if (serializedIndex == -1 && tx_state.hashBlock == ABANDON_HASH) {
+            tx_state.nIndex = 0;
+            setAbandoned();
+        } else if (serializedIndex == -1) {
+            tx_state.nIndex = -1;
```

Comment:
Can we set 0 here, or change MarkConflicted to set -1? I think it'd be good to be consistent about what conflicted transactions look like.

Reply:
You're right that an inconsistency from my part. Corrected.